File: python-sdk/test/t_file.py
# -*- coding: utf-8 -*-
import unittest
from test.global_test_config import oauth
from fangcloudsdk.client import Client


class T_File(unittest.TestCase):
    test_copy_id = None

    def setUp(self):
        oauth.refresh_token = "bceecd8b-ba45-4aca-8c4e-a80ed253f9f2"
        # oauth.access_token = "94a39c41-25c8-4096-92d9-5c3af3d10138"
        self.client = Client(oauth)
        self.test_file_id = 501000524950
        self.test_target_folder_id = 501000031450

    def test_01_get_file_info(self):
        res = self.client.file(self.test_file_id).info()
        self.assertTrue(res['success'])

    def test_02_file_update(self):
        res = self.client.file(self.test_file_id).update("dog.png", "xuning's update")
        self.assertTrue(res['success'])

    def test_03_delete_file(self):
        res = self.client.file().delete(file_ids=[self.test_file_id])
        self.assertTrue(res['success'])

    # def test_delete_file_from_trash(self):
    #     res = self.client.file().delete_from_trash(file_ids=None)
    #     self.assertTrue(res['success'])

    def test_04_recovery_file_from_trash(self):
        res = self.client.file().recovery_from_trash(file_ids=[self.test_file_id])
        self.assertTrue(res['success'])

    def test_05_move_file(self):
        res = self.client.file().move(file_ids=[self.test_file_id], target_folder_id=self.test_target_folder_id)
        self.assertTrue(res['success'])

    def test_06_re_move_file(self):
        res = self.client.file().move(file_ids=[self.test_file_id], target_folder_id=0)
        self.assertTrue(res['success'])

    def test_07_upload(self):
        res = self.client.file().upload(parent_id=self.test_target_folder_id, name="dog.png")
        self.assertTrue(res['success'])

    def test_08_new_version(self):
        res = self.client.file(self.test_file_id).new_version(name="dog.png", remark="xuning, new version test")
        self.assertTrue(res['success'])

    def test_09_file_download(self):
        res = self.client.file(self.test_file_id).download()
        self.assertTrue(res['success'])

    def test_10_preview_file(self):
        res = self.client.file(self.test_file_id).preview(kind="image_64")
        self.assertTrue(res['success'])

    def test_11_preview_download(self):
        res = self.client.file(self.test_file_id).preview_download(page_index=0, kind="image_64")

    def test_12_copy(self):
        res = self.client.file(self.test_file_id).copy(target_folder_id=self.test_target_folder_id)
        T_File.test_copy_id = res['new_file']['id']
        self.assertTrue(res['success'])

    def test_13_re_copy(self):
        res = self.client.file().delete(file_ids=[self.test_copy_id])
        self.assertTrue(res['success'])

    def test_14_upload(self):
        res = self.client.file().upload(parent_id=self.test_target_folder_id, name="dog.png")
        self.assertTrue(res['success'])

    def test_15_new_version(self):
        res = self.client.file(self.test_file_id).new_version(name="dog.png", remark="xuning, new version test")
        self.assertTrue(res['success'])

    def test_16_file_download(self):
        res = self.client.file(self.test_file_id).download()
        self.assertTrue(res['success'])

    def test_17_preview_file(self):
        res = self.client.file(self.test_file_id).preview(kind="image_64")
        self.assertTrue(res['success'])

    def test_18_preview_download(self):
        res = self.client.file(self.test_file_id).preview_download(page_index=0, kind="image_64")
        self.assertTrue(res['success'])

    def tearDown(self):
        pass

File: php-sdk/test/t_file.php
<?php
use PHPUnit\Framework\TestCase;

require_once "test_global_var.php";
require_once "../fangcloudsdk/Client.class.php";

/**
 * Created by PhpStorm.
 * User: xuning
 * Date: 16-10-12
 * Time: 下午12:58
 */
class t_file extends TestCase {

    public $oauth;
    public $client;
    public $test_file_id = 501000524950;
    public $test_target_folder_id = 501000005400;

    public function __construct() {
        $this->oauth = test_global_var::get_test_oauth();
        $this->client = new Client($this->oauth);
    }

    public function test_info() {
        $res = $this->client->File($this->test_file_id)->info();
        TestCase::assertEquals(true, $res['success']);
    }

    public function test_update() {
        $res = $this->client->File($this->test_file_id)->update("dog.png", "xuning test update");
        TestCase::assertEquals(true, $res['success']);
    }

    public function test_delete() {
        $res = $this->client->File()->delete(array($this->test_file_id));
        TestCase::assertEquals(true, $res['success']);
    }

    public function test_delete_from_trash() {
        $res = $this->client->File()->delete_from_trash(array($this->test_file_id));
        TestCase::assertEquals(true, $res['success']);
    }

    public function test_recovery_from_trash() {
        $res = $this->client->File()->recovery_from_trash(array($this->test_file_id));
        TestCase::assertEquals(true, $res['success']);
    }

    public function test_move() {
        $res = $this->client->File()->move(array($this->test_file_id), $this->test_target_folder_id);
        TestCase::assertEquals(true, $res['success']);
    }

    public function test_new_version() {
        $res = $this->client->File($this->test_file_id)->new_version("dog,png", "test_new_version");
        TestCase::assertEquals(true, $res['success']);
    }

    public function test_upload() {
        $res = $this->client->File()->upload($this->test_target_folder_id, "dog.png");
        TestCase::assertEquals(true, $res['success']);
    }

    public function test_download() {
        $res = $this->client->File($this->test_file_id)->download();
        TestCase::assertEquals(true, $res['success']);
    }

    public function test_preview() {
        $res = $this->client->File($this->test_file_id)->preview(false, "image_64");
        TestCase::assertEquals(true, $res['success']);
    }

    public function test_preview_download() {
        $res = $this->client->File($this->test_file_id)->preview_download(1, "image_64");
        TestCase::assertEquals(true, $res['success']);
    }

    public function test_copy() {
        $res = $this->client->File()->copy($this->test_file_id, $this->test_target_folder_id);
        TestCase::assertEquals(true, $res['success']);
    }

}